packages/server/api/src/app/workers/queue/job-queue.ts

```diff
@@ -64,35 +65,42 @@ export const jobQueue = (log: FastifyBaseLogger): QueueManager => ({
         }, '[redisQueue#removeRepeatingJob] removing the jobs')
         await queue.removeJobScheduler(flowVersionId)
     },
-
 })
 
-async function ensureQueueExists(queueName: QueueName): Promise<Queue> {
+async function ensureQueueExists(queueName: QueueName, log?: FastifyBaseLogger): Promise<Queue> {
     if (!isNil(bullMqGroups[queueName])) {
         return bullMqGroups[queueName]
     }
     const isOtpEnabled = system.getBoolean(AppSystemProp.OTEL_ENABLED)
-    bullMqGroups[queueName] = new Queue(
-        queueName,
-        {
-            telemetry: isOtpEnabled ? new BullMQOtel(queueName) : undefined,
-            connection: await redisConnections.createNew(),
-            defaultJobOptions: {
-                attempts: 5,
-                backoff: {
-                    type: 'exponential',
-                    delay: EIGHT_MINUTES_IN_MILLISECONDS,
-                },
-                removeOnComplete: true,
-                removeOnFail: {
-                    age: REDIS_FAILED_JOB_RETENTION_DAYS,
-                    count: REDIS_FAILED_JOB_RETRY_COUNT,
-                },
-            },
 
+    const options = {
+        telemetry: isOtpEnabled ? new BullMQOtel(queueName) : undefined,
+        connection: await redisConnections.createNew(),
+        defaultJobOptions: {
+            attempts: 5,
+            backoff: {
+                type: 'exponential',
+                delay: EIGHT_MINUTES_IN_MILLISECONDS,
+            },
+            removeOnComplete: true,
+            removeOnFail: {
+                age: REDIS_FAILED_JOB_RETENTION_DAYS,
+                count: REDIS_FAILED_JOB_RETRY_COUNT,
+            },
         },
-    )
+
+    }
+
+    bullMqGroups[queueName] = new Queue(queueName, options)
     await bullMqGroups[queueName].waitUntilReady()
+    
+    if (log) {
+        const queueEvents = new QueueEvents(queueName, options)
+        await queueEvents.waitUntilReady()
+        queueMetrics(log, queueEvents).detach()
```

Comment:
Make log not optional, If you ever couldn't pass it use (system.globalLogger()), it would be great if you can pass it from top.

Let's remove bullMqGroups actually because we only have one queue and queueEvents, previously we had like 5 queues that was removed.